Userland/Libraries/LibWeb/Infra/Strings.cpp

```diff
@@ -144,4 +144,27 @@ ErrorOr<String> to_ascii_uppercase(StringView string)
     return string_builder.to_string();
 }
 
+// https://infra.spec.whatwg.org/#isomorphic-encode
+ByteBuffer isomorphic_encode(StringView input)
+{
+    ByteBuffer buf = {};
+    for (auto code_point : Utf8View { input }) {
+        // VERIFY(code_point <= 0xFF);
+        if (code_point > 0xFF)
+            dbgln("FIXME: Trying to isomorphic encode a string with code points > U+00FF.");
+        buf.append((u8)code_point);
+    }
+    return buf;
+}
+
+// https://infra.spec.whatwg.org/#isomorphic-decode
+String isomorphic_decode(ReadonlyBytes input)
+{
+    StringBuilder builder(input.size());
+    for (u8 code_point : input) {
+        builder.append_code_point(code_point);
+    }
+    return builder.to_string_without_validation();
```

Comment:
I don't think this is the right thing to do - `String` is by definition UTF-8, but here we can be throwing any encoding into the string. The `to_string_without_validation` isn't a path to avoid that invariant, rather it's a performance optimization for when we already know the data going into `String` is UTF-8 (as validation isn't free).

We use `ByteString` to represent strings of arbitrary encodings.

Reply:
Unless I'm misunderstanding what `append_code_point` does that's exactly what I'm doing. This function takes ISO-8859-1 encoded bytes and converts them to the corresponding UTF-8 encoding.

Reply:
See also: `Latin1Decoder::process` and `Decoder::to_utf8`

Reply:
Ah I see, I think I was misunderstanding what isomorphic decoding means. But yes, this looks right. Thanks :)